src/zcash/address/orchard.hpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2021 The Zcash Developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#ifndef ZCASH_ADDRESS_ORCHARD_H
+#define ZCASH_ADDRESS_ORCHARD_H
+
+#include "streams.h"
+#include "rust/orchard/keys.h"
+
+namespace libzcash {
+
+class OrchardRawAddress
+{
+private:
+public: //FIXME
+    std::unique_ptr<OrchardRawAddressPtr, decltype(&orchard_address_free)> inner;
+
+    OrchardRawAddress() : inner(nullptr, orchard_address_free) {}
+
+    OrchardRawAddress(OrchardRawAddressPtr* ptr) : inner(ptr, orchard_address_free) {}
```

Comment:
Where would this constructor be used?

Reply:
This is an artifact of some headaches I had around handling persistence of note metadata; it's currently used in my wallet implementation branch in https://github.com/nuttycom/zcash/blob/kjn/wallet_orchard-add_orchard/src/wallet/orchard.cpp#L63 - I'm not sure if this will be the ultimate design though, so I'll pull out the constructor from this PR. It's gross, I agree.

Reply:
It is needed for `OrchardIncomingViewingKey::Address`